server/rest/src/main/java/org/infinispan/rest/resources/HealthCheckResource.java

```diff
@@ -0,0 +1,58 @@
+package org.infinispan.rest.resources;
+
+import static org.infinispan.rest.framework.Method.GET;
+import static org.infinispan.rest.framework.Method.HEAD;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+import org.infinispan.manager.CacheManagerInfo;
+import org.infinispan.manager.DefaultCacheManager;
+import org.infinispan.rest.InvocationHelper;
+import org.infinispan.rest.NettyRestResponse;
+import org.infinispan.rest.framework.ResourceHandler;
+import org.infinispan.rest.framework.RestRequest;
+import org.infinispan.rest.framework.RestResponse;
+import org.infinispan.rest.framework.impl.Invocations;
+
+import io.netty.handler.codec.http.HttpResponseStatus;
+
+public class HealthCheckResource implements ResourceHandler {
+
+   private final InvocationHelper helper;
+
+   public HealthCheckResource(InvocationHelper helper) {
+      this.helper = helper;
+   }
+
+   @Override
+   public Invocations getInvocations() {
+      return new Invocations.Builder()
+
+            .invocation().methods(GET, HEAD).anonymous(true).path("/health/live")
+            .handleWith(this::notifyServerRunning)
+
+            .invocation().methods(GET, HEAD).anonymous(true).path("/health/ready")
+            .handleWith(this::verifyServerReady)
+
+            .create();
+   }
+
+   public CompletionStage<RestResponse> notifyServerRunning(RestRequest request) {
+      NettyRestResponse.Builder builder = helper.newResponse(request);
+      return CompletableFuture.completedFuture(builder.status(HttpResponseStatus.OK).build());
```

Comment:
So this is more like a PONG than anything else as it is always OK?

Reply:
Correct, just to reply that the process is alive.

I believe this one will make more sense once we get to #14803. The server would start ASAP, and the live/start probe here would reply OK, and while the cache manager hasn't started, the ready probe would return 503.